appium/sample-scripts/python/device_finder.py

```diff
@@ -23,50 +30,66 @@ def _build_headers(self, headers=None):
 
     """ Returns list of devices
     """
-    def get_devices(self, limit=0):
-        return self.get("devices?limit=%s" % (limit))
+    def get_devices(self, limit=0, query_param=""):
+        # https://cloud.testdroid.com/api/v2/devices?filter=b_online_eq_true;b_locked_eq_false;b_enabled_eq_true;n_creditsPrice_eq_0;s_osType_eq_ios
+        query_str = "devices?limit=%s" % (limit)
+        if self.device_group:
+            query_str += "&device_group_id[]=%s" % (self.device_group)
+        # add extra query params, defaults to all
+        query_str += "&filter=b_online_eq_true;b_locked_eq_false;b_enabled_eq_true"
```

Comment:
Need to confirm that this query filters correctly. Currently the 'locked' filter doesn't seem to work.